### ISSUE TITLE
Blazor WEB Assembly reloads when using NavigationManager.NavigateTo()

### ISSUE THREAD

**Issue by sroy2020** [dotnet/aspnetcore#26015]
### Describe the bug
Blazor WEB Assembly reloads when using `NavigationManager.NavigateTo("/Search")`. The `@page "/Search"` is in a different Razor file than the one where `@page "/"` is specified although the code is VERY similar. Using something like the following `NavigationManager.NavigateTo("/Transfer/" + Transfer.Id + "/" + cacheId.ToString());` does not make the WASM to reload.

Also using `NavigationManager.NavigateTo("/")` will reloads the assembly.

Doing something very similar with the Blazor WASM demo page works fine. It does not reload at all even with a line `NavigationManager.NavigateTo("/")`.

Any clue what could be wrong?

Thanks

### To Reproduce
See description

Here's my App.razor (nothing special):
```
<Router AppAssembly="@typeof(Program).Assembly">
    <Found Context="routeData">
        <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)" />
    </Found>
    <NotFound>
        <div class="d-flex justify-content-center">
            <div class="page-not-found border-left border-bottom mt-lg-5"></div>
        </div>
    </NotFound>
</Router>
```

### Exceptions (if any)
None

### Further technical details
.NET Core SDK (reflecting any global.json):
 Version:   3.1.400
 Commit:    035fb2aa2f

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.400\

Host (useful for support):
  Version: 3.1.6
  Commit:  3acd9b0cd1

.NET Core SDKs installed:
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.400 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

![VS](https://user-images.githubusercontent.com/61807678/93496352-16129980-f8dd-11ea-83f2-e3fba8bd1102.png)
![Packages](https://user-images.githubusercontent.com/61807678/93496940-d7311380-f8dd-11ea-802a-0989d2bc2845.png)

**Comment by mkArtakMSFT** [comment]
Thank you for filing this issue. In order for us to investigate this issue, please provide a minimalistic repro project (ideally a GitHub repo) that illustrates the problem.

**Comment by mkArtakMSFT** [comment]
Also, can you please try out the 5.0 RC1 release and see whether the issue is resolved there?

**Comment by LaughingJohn** [comment]
Just a thought but if your navigation is triggered by a button in a form that will "submit" the form and cause the page to reload. Adding type="button" stops that happening.

**Comment by sroy2020** [comment]
Hi,

I'm still trying to migrate to 5.0 RC1 unfortunately I'm having really strange errors even though I';m following the instruction from [ASP.NET Core updates in .NET 5 Release Candidate 1](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-5-release-candidate-1/).

I also reviewed the code triggering the navigation it's not part a form with a submit.

`    <div class="float-right mr-2 cursor-pointer" @onclick="@(() => Close())" data-toggle="tooltip" data-placement="top" title="@Localizer.Close">🗙</div>`

I'll continue trying to migrate to 5.0 RC1.

Thanks

**Comment by sroy2020** [comment]
Hi,

Before trying again to migrate to 5.0 RC1, I tried the following. I replaced the line calling `NavigationManager.NavigateTo("/Transfer/" + Transfer.Id + "/" + cacheId.ToString());` since it's not any problem with NavigationManager.NavigateTo("/Search"); and also tried with `NavigationManager.NavigateTo("/");`.

There's no reload, so I guess I'll have to investigate that route, i.e. the "navigation context" between the good one and the bad one.

Thanks
